### ISSUE TITLE
Cheap inversion with 32 bytes helper value

### ISSUE THREAD

**Issue by kilic** [daimo-eth/p256-verifier#33]
Alongside with pubkey and signature transaction caller may provide an inverted value then we can check the inversion `(a * a_inv == 1) mod n`. So that modexp call can be skipped

**Comment by nalinbhardwaj** [comment]
Good idea, while this would certainly reduce computation gas cost, it would increase call data used and thus, the L1 calldata fee (which is the dominant cost) for users of the contract on L2s; Since that's the primary use case ATM we probably won't implement this.

**Comment by kilic** [comment]
Single 32 byte helper would suffice If it is multiple signature verification case with batch inversion technique. But I'm not sure it would worth to touch too many places if modexp cost is not that high
